File: Prueba/PruebaI/MongoConnection.py
from pymongo import MongoClient
from University import University  # Importa tu clase University correctamente

class MongoConnection:  # Usa mayúscula correctamente
    def __init__(self):
        user = "isaac"
        password = "isaac"  # Cambia aquí tu contraseña si corresponde
        cluster = "cluster0.xaitfht"
        uri = f"mongodb+srv://{user}:{password}@{cluster}.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0"
        self.client = MongoClient(uri)
        self.db = self.client["University"]  # Nombre exacto de tu base

    def get_universities(self):
        collection = self.db["university"]  # Nombre exacto de tu colección
        documents = collection.find()
        universities = []
        for doc in documents:
            u = University(
                doc.get("Id"),
                doc.get("name"),
                doc.get("monthlyIncome"),
                doc.get("numberOfStudent")
            )
            universities.append(u)
        return universities

    def close(self):
        self.client.close()

File: Exam2/src/main/java/ec/espe/edu/utils/MongoConnection.java

package ec.espe.edu.utils;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;

/**
 *
 * @author Isaac Maisincho Crafters_Market DCCO ESPE
 */
public class MongoConnection {
     private static final String URI = "mongodb+srv://isaac:isaac@cluster0.xaitfht.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0";
    private static final String DATABASE_NAME = "Soccer";
    
    private static MongoClient mongoClient = null;

    private MongoConnection() {
      
    }

    public static MongoDatabase getDatabase() {
        if (mongoClient == null) {
            mongoClient = MongoClients.create(URI);
        }
        return mongoClient.getDatabase(DATABASE_NAME);
    }

    public static void closeConnection() {
        if (mongoClient != null) {
            mongoClient.close();
            mongoClient = null;
        }
    }
    
    }